### PR TITLE
Fix to user names after non-messages

### PR DESCRIPTION
**Pull request by dpatti**
In #545, a change was made that made timestamps show up correctly when non-chat
messages were entered before the chat was. As an unintended side-effect, the
user names were not being displayed after a non-chat message if that user spoke
before it, which made for some odd dialog.

To be more structured, all message appending was separated to a single function
regardless of message. There it is checked to see whether a new date header is
necessary or not. Now we have accurate date headers and user names appearing
when they should be.

Note that while no prepending code was touched, as it is not affected, it might
be something to refactor later.

### CODE REVIEW

File: JabbR/Chat.ui.js [hunk 1]
```diff
@@ -1606,6 +1603,21 @@
 
             return $element;
         },
+        appendMessage: function (newMessage, room) {
+            // Determine if we need to show a new date header
+            if (!$(newMessage).is('.date-header')) {
+                var lastMessage = room.messages.find("li[data-timestamp]").last(),
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: single quotes

**Comment by davidfowl** [comment]
Can you show what was broken and what this fixes? (visually)

**Comment by dpatti** [comment]
Missing name

The second message here should be displaying a name for readability.

**Comment by dpatti** [comment]
Here's another improvement: before date headers only showed before chat messages. As you can see here, a non-chat message happened when a date header should have been inserted. This was the primary reason for creating `appendMessage`.

No date header

**Comment by davidfowl** [comment]
Since these changes seem to be regressing other features we'll need to find a way to add tests for these things.

**Pull request merged**